### PR TITLE
fix: color visualizer does not affect body element's background

### PR DESCRIPTION
**Pull request by ThanyaLeif**
#### Details

This PR address #4615 
Before this PR, Accessibility Insights for Web's Color ad-hoc visualizer, does not work against background colors / images applied to a page's body element.

I changed the viewer selector so that the properties are applied to the entire document. This also fixes the bug related to assessment 16.1
Below is the behavior of the ad-hoc color visualizer before and after the changes

The first image shows a page with a gradient background, which is not affected by the visualizer.
image

The second image shows the entire page in grayscale, after the changes were applied.
image

The third image shows how the color rule is also applied from assessment 16.1
image

##### Motivation

Address #4615 

##### Context
For some reason it seems that the 'body' selector only affects its children, not the body itself.

#### Pull request checklist

- [x] Addresses an existing issue: #4615 
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/injected/analyzers/filter-results.ts [hunk 1]
```diff
@@ -10,7 +10,8 @@ export const filterNeedsReviewResults = (results: ScanResults): ScanResults => {
         v =>
             v.id !== 'aria-input-field-name' &&
             v.id !== 'color-contrast' &&
-            v.id !== 'th-has-data-cells',
+            v.id !== 'th-has-data-cells' &&
+            v.id !== 'label-content-name-mismatch',
```
Comment (dbjorge) [inline src/injected/analyzers/filter-results.ts:14]: Did you intend to include the `filter-results` change (and its tests) here? I suspect they're intended to be for #4339, not this PR

File: src/scanner/custom-rules/color-rule.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export const colorConfiguration: RuleConfiguration = {
     ],
     rule: {
         id: 'select-body',
-        selector: 'body',
+        selector: '*',
```
Comment (dbjorge) [inline src/scanner/custom-rules/color-rule.ts:16]: I think the `select-body` id is misleading now that this is selecting more than just the body element. Could you update the ID to something that reflects the new selector?

File: src/injected/analyzers/filter-results.ts (hunk 1 above)
Comment (ThanyaLeif) [reply src/injected/analyzers/filter-results.ts]: Right, this belongs to #4339
I undid changes to this file and `src/tests/unit/tests/scanner/custom-rules/color-rule.test.ts` in my last commit.

File: src/scanner/custom-rules/color-rule.ts (hunk 2 above)
Comment (ThanyaLeif) [reply src/scanner/custom-rules/color-rule.ts:16]: Done
I renamed this and other related ids in my last commit.

Comment (dbjorge) [reply src/scanner/custom-rules/color-rule.ts:16]: Thanks! However, in this context, the term "document" has a specific meaning which is still different from what's being selected here. Maybe `select-each-element` or `select-all-elements`?

Comment (waabid) [reply src/scanner/custom-rules/color-rule.ts:16]: I would also extract the name out to a variable so you can reference that in `use-of-color.tsx` and `visualization.tsx`.

File: src/scanner/custom-rules/color-rule.ts [hunk 3]
```diff
@@ -12,8 +12,8 @@ export const colorConfiguration: RuleConfiguration = {
         },
     ],
     rule: {
-        id: 'select-body',
-        selector: 'body',
+        id: 'select-document',
```
Comment (waabid) [inline src/scanner/custom-rules/color-rule.ts:15]: While I think this solution works, I think a simpler solution may suffice. With regards to both uses of this rule, we only add the css styles to a single/first element in the `SingleTargetDrawer` (line 47 onwards `single-target-drawer.ts`).

In the webpage examples given, I believe the first element returned is actually the `html` element, so if we change the selector to `html` we should get the same result (and can also rename to `select-html`, per Dan's suggestion).

**Review by waabid — CHANGES_REQUESTED** [review]
Left a comment.

File: src/scanner/custom-rules/color-rule.ts (hunk 2 above)
Comment (ThanyaLeif) [reply src/scanner/custom-rules/color-rule.ts:16]: Makes sense to me
Is there a specific file in the project where we put those common variables @waabid ?

Comment (waabid) [reply src/scanner/custom-rules/color-rule.ts]: I think the `color-rule.tsx` file should be fine for this purpose in this case.

**Review by dbjorge — APPROVED** [review]

**Review by waabid — APPROVED** [review]
🚢

**Pull request merged**